lib/kitchen/terraform/command/validate.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Copyright 2016 New Context Services, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "kitchen"
+require "kitchen/terraform/command_flag/color"
+require "kitchen/terraform/command_flag/variable_files"
+require "kitchen/terraform/command_flag/variables"
+require "kitchen/terraform/shell_out"
+
+module Kitchen
+  module Terraform
+    module Command
+      # Validate is the class of objects which represent the <tt>terraform validate</tt> command.
+      class Validate
+        class << self
+          # Initializes an instance by running `terraform validate`.
+          #
+          # @param options [::Hash] the command options.
+          # @option options [true, false] :color a toggle for colored output.
+          # @option options [::String] :directory the directory in which to run the command.
+          # @option options [::Integer] :timeout the maximum duration in seconds to run the command.
+          # @option options [::Array<::String>] :variable_files files containing variables for the configuration.
+          # @option options [::Hash{::String => ::String}] :variables variables for the configuration.
+          # @raise [::Kitchen::Terraform::Error] if the result of running the command is a failure.
+          # @return [self]
+          # @yieldparam validate [::Kitchen::Terraform::Command::Validate] an instance initialized with the output of
+          #   the command.
+          def run(options)
+            new(options).tap do |validate|
+              ::Kitchen::Terraform::ShellOut.run(
+                command: validate,
+                directory: options.fetch(:directory),
+                timeout: options.fetch(:timeout),
+              )
+              yield validate: validate if block_given?
+            end
+
+            self
+          end
+        end
+
+        def ==(validate)
+          to_s == validate.to_s
+        end
+
+        def store(output:)
+          @output = output
+
+          self
+        end
+
+        def to_s
+          ::Kitchen::Terraform::CommandFlag::Variables.new(
+            command: ::Kitchen::Terraform::CommandFlag::VariableFiles.new(
+              command: ::Kitchen::Terraform::CommandFlag::Color.new(
+                command: ::String.new("terraform validate -check-variables=true"),
```

Comment:
Looks like this would be the place where flag should be specifiable or version specific

Reply:
Yes, `CommandFlag::Validate` needs to be defined, similar to `CommandFlag::Upgrade`. Additionally, `Command::Validate` needs to be enhanced to check the value of `Command::Version` and then defer to a version specific implementation. `Command::Version` should also be enhanced to cache the first instance.